cpp/src/merge/merge.cu

```diff
@@ -0,0 +1,329 @@
+
+#include <thrust/execution_policy.h>
+#include <thrust/for_each.h>
+#include <thrust/iterator/constant_iterator.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <thrust/tuple.h>
+#include <thrust/device_vector.h>
+#include <thrust/merge.h>
+#include <algorithm>
+#include <utility>
+#include <vector>
+#include <memory>
+#include <nvstrings/NVCategory.h>
+
+#include <cudf/cudf.h>
+#include <cudf/types.hpp>
+#include <cudf/copying.hpp>
+#include <cudf/table.hpp>
+#include "table/device_table.cuh"
+#include "table/device_table_row_operators.cuh"
+#include "bitmask/bit_mask.cuh"
+#include "string/nvcategory_util.hpp"
+#include "rmm/thrust_rmm_allocator.h"
+#include "utilities/cuda_utils.hpp"
+
+namespace {
+
+using bit_mask::bit_mask_t;
+
+/**
+ * @brief Source table identifier to copy data from.
+ */
+enum side_value { LEFT_TABLE = 0, RIGHT_TABLE };
```

Comment:
This is a good place to use an `enum class` backed by a `bool`. Since you only need to store 2 values, make that explicit by using a `bool`. 
```suggestion
enum class side : bool { LEFT, RIGHT };
```